### PR TITLE
Rename distro archive folder to camunda-cloud-zeebe

### PR DESCRIPTION
**Pull request by menski**
## Description

- Rename folder inside distro archive from `zeebe-broker` to `camunda-cloud-zeebe`
- Align with archive name and Operate/Tasklist
- zeebe-broker is misleading as the bin folder also contains the gateway and the zbctl to run

## Related issues

related #6913

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: dist/pom.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
   <modelVersion>4.0.0</modelVersion>
   <name>Zeebe Distribution</name>
-  <artifactId>camunda-cloud-zeebe</artifactId>
+  <artifactId>zeebe-distribution</artifactId>
```
Comment (npepinpe) [inline dist/pom.xml:5]: Why revert the name of the module?

**Review by npepinpe — CHANGES_REQUESTED** [review]

File: dist/pom.xml (hunk 1 above)
Comment (menski) [reply dist/pom.xml:5]: I thought it's not needed, but learned today in the Operate/Tasklist release that it is required. I will revert this change

**Review by npepinpe — APPROVED** [review]
:+1:

**Pull request merged**